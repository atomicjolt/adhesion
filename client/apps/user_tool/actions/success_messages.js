import wrapper from 'atomic-fuel/libs/constants/wrapper';

// Local actions
const actions = ['CLEAR_SUCCESS_MESSAGES'];

// Actions that make an api request
const requests = [];

export const Constants = wrapper(actions, requests);

export const clearSuccessMessages = () => ({
  type: Constants.CLEAR_SUCCESS_MESSAGES,
});
