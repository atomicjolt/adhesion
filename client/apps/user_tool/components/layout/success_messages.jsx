import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { clearSuccessMessages } from '../../actions/success_messages';

const select = state => ({ messages: state.successMessages });

export class SuccessMessages extends React.Component {
  static propTypes = {
    clearSuccessMessages: PropTypes.func.isRequired,
    messages: PropTypes.array.isRequired,
  }

  componentDidUpdate() {
    const { clearSuccessMessages:clear, messages } = this.props;

    if (_.isEmpty(messages)) { return; }

    setTimeout(() => {
      clear();
    }, 4000);
  }

  render() {
    const { messages } = this.props;

    const renderedMessages = _.map(messages,
      (message, index) => <li key={index}>{message}</li>
    );

    return (
      <div id="success-messages" aria-live="polite">
        { !_.isEmpty(renderedMessages) && (
          <div className="messages success">
            <ul>{renderedMessages}</ul>
          </div>
        )}
      </div>
    );
  }
}

export default connect(select, { clearSuccessMessages })(SuccessMessages);
