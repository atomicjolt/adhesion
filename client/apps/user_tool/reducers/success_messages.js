import { Constants as ApplicationConstants } from '../actions/application';
import { Constants as SuccessMessagesConstants } from '../actions/success_messages';

const initialState = () => ([]);

export default (state = initialState(), action) => {
  switch (action.type) {
    case SuccessMessagesConstants.CLEAR_SUCCESS_MESSAGES: {
      return [];
    }

    case ApplicationConstants.UPDATE_ACCOUNT_USER_DONE: {
      if (!action.error) {
        return [...state, 'User updated successfully.'];
      }

      return state;
    }

    default:
      return state;
  }
};
