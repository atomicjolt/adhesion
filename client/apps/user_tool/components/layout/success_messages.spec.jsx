import React from 'react';
import { shallow } from 'enzyme';

import { SuccessMessages } from './success_messages';

describe('SuccessMessages', () => {
  const props = {
    messages: ['User updated successfully.'],
  };

  it('renders the messages', () => {
    const successMessages = shallow(<SuccessMessages
      messages={props.messages}
    />);

    expect(successMessages).toMatchSnapshot();
  });

  describe('when there are no messages', () => {
    it('renders nothing', () => {
      const successMessages = shallow(<SuccessMessages
        messages={[]}
      />);

      expect(successMessages.html()).toEqual(null);
    });
  });
});
