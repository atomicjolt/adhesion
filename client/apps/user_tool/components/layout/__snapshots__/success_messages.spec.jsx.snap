// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessages renders the messages 1`] = `
<div
  aria-live="polite"
  id="success-messages"
>
  <div
    className="messages success"
  >
    <ul>
      <li
        key="0"
      >
        User updated successfully.
      </li>
    </ul>
  </div>
</div>
`;

exports[`SuccessMessages when there are no messages only renders the outer div 1`] = `
<div
  aria-live="polite"
  id="success-messages"
/>
`;
